Serializers with many=True should return empty list rather than empty dict

A serializer with `many=True` returns a list of dicts when accessing `serializer.errors` or `serializer.validated_data`. If these are empty, however, an empty dict is returned (just as when `many=False`). I believe that it would be more consistent to return an empty list in this case.

Example test case: https://github.com/tomchristie/django-rest-framework/compare/tomchristie:master...nip3o:empty-errors?expand=1

    ==================================================================================== FAILURES ====================================================================================
    _______________________________________________________________ BulkCreateSerializerTests.test_bulk_create_errors ________________________________________________________________
    tests/test_serializer_bulk_update.py:80: in test_bulk_create_errors
        self.assertEqual(serializer.validated_data, [])
    E   AssertionError: {} != []
    _______________________________________________________________ BulkCreateSerializerTests.test_bulk_create_success _______________________________________________________________
    tests/test_serializer_bulk_update.py:49: in test_bulk_create_success
        self.assertEqual(serializer.errors, [])
    E   AssertionError: {} != []
    ================================================================ 2 failed, 743 passed, 2 warnings in 7.98 seconds ===============================================================

--- a/rest_framework/serializers.py
+++ b/rest_framework/serializers.py
@@ -667,6 +667,28 @@ class ListSerializer(BaseSerializer):
 
         return self.instance
 
+    def is_valid(self, raise_exception=False):
+        # This implementation is the same as the default,
+        # except that we use lists, rather than dicts, as the empty case.
+        assert hasattr(self, 'initial_data'), (
+            'Cannot call `.is_valid()` as no `data=` keyword argument was '
+            'passed when instantiating the serializer instance.'
+        )
+
+        if not hasattr(self, '_validated_data'):
+            try:
+                self._validated_data = self.run_validation(self.initial_data)
+            except ValidationError as exc:
+                self._validated_data = []
+                self._errors = exc.detail
+            else:
+                self._errors = []
+
+        if self._errors and raise_exception:
+            raise ValidationError(self.errors)
+
+        return not bool(self._errors)
+
     def __repr__(self):
         return unicode_to_repr(representation.list_repr(self, indent=1))
 

--- a/tests/test_serializer_bulk_update.py
+++ b/tests/test_serializer_bulk_update.py
@@ -46,6 +46,7 @@ class BulkCreateSerializerTests(TestCase):
         serializer = self.BookSerializer(data=data, many=True)
         self.assertEqual(serializer.is_valid(), True)
         self.assertEqual(serializer.validated_data, data)
+        self.assertEqual(serializer.errors, [])
 
     def test_bulk_create_errors(self):
         """
@@ -76,6 +77,7 @@ class BulkCreateSerializerTests(TestCase):
         serializer = self.BookSerializer(data=data, many=True)
         self.assertEqual(serializer.is_valid(), False)
         self.assertEqual(serializer.errors, expected_errors)
+        self.assertEqual(serializer.validated_data, [])
 
     def test_invalid_list_datatype(self):
         """

